### ISSUE TITLE
Navigation, performance and overflow issues on larger datasets

### ISSUE THREAD

**Issue by hrumhurum** [retypeapp/retype#253]
Retype is a cool tool for smaller websites, but its limits are pushed quite easily with a bit more involved workloads.

Here is an example of a medium workload of that kind:

https://boring-raman-80b722.netlify.app/gapotchenko.fx/reference/gapotchenko.fx.threading.tests/threadingcapabilitiestests/methods/threadingcapabilities_logicalprocessorcount/

1. As you can see, the page title is not getting wrapped and it is overflowing to the right side of the screen when viewed in a maximized browser window. It would be nice if Retype could do a sane thing there, like forcing an overflow even when there are no line-breaking opportunities available

2. The navigation pane gets wrecked by longer titles and deeper hierarchy

3. The performance of page loads are suffering quite a bit. Try to click on some items in navigation pane and note how relatively slow every page load is. We checked that and it seems that the delay is CPU bound somewhere deep inside the javascript

Are those expectations aligned with the sprit of the product? If yes then we would be more than happy to become your long-term commercial customer.

**Comment by geoffreymcgill** [comment]
Hi @hrumhurum. Thanks for the feedback. 

Just for clarification, it appears the issues you have noted are related to the `api` doc generation for C# projects and especially for large .NET Solutions. Not general content pages generated from .md files. 

We have have the same experience generating docs on a mega large C# project, and it's frustrating. We pushed the limits, much like your project seems to have. This is also one of the major reasons we have not actively promoted the C# api doc generation. We're just not happy with the client performance in the browser.

A few of the rendering issues are relatively easy fixes, although for a while our priority has been allocated to other features. I can see about bumping up the priority and making some easy wins.

There are a few issues though that are not easy fixes. We have spent piles of time trying to improve the performance, but VueJS just hits a functional limit and the client performance takes a hit. The generation of the pages is still super fast from the RetypeApp side but running the finished pages and site in the browser uncovers performance issues.

Because these performance issues have been almost exclusively related to the `api` generation and because we haven't had a lot of demand for that functionality, we have just had to put the work to improve the performance on hold.

I would love to jump back into this but we're a small team and working out these C# `api` doc performance issues is going to require more resources than we currently have available. I wish I had better news, a bigger budget, and about 5 more engineers.

The performance issue with your point #3 is super annoying to me too. Even navigating to small pages there is a noticeable delay. We are going to investigate and try to figure that one out.

Hope this helps.

**Comment by geoffreymcgill** [comment]
okay, hold on a second. Those are not Retype `api` generated pages. Those are all `.md` pages that must have been made with another tool.

Yep, then we have a performance issue and a few other UI issues. 

We will be investigating right away, especially the navigation delay.

**Comment by hrumhurum** [comment]
Hi @geoffreymcgill,

Yes, the `.md` files are generated by a draft version of a custom tool.

The source archive with those `.md` files can be useful for issue reproduction purposes: https://boring-raman-80b722.netlify.app/gapotchenko.fx/source/docs-gapotchenko-com-gapotchenko-fx.zip

Thank you for paying the attention to those details. Retype is a rare find and I am already sold.

**Comment by geoffreymcgill** [comment]
The issue with the long wrapping of page titles and navigation labels has been fixed. The fix will be included in the next release of Retype. We will publish a `-preview` release before the final version. Once that `-preview` is available, it would be great to get your feedback on the fix. 

We haven't had a chance to review the performance issue but it is a super high priority item.

**Comment by geoffreymcgill** [comment]
### Before with Retype v1.11

<img width="320" alt="Screen Shot 2022-02-02 at 10 06 31 PM" src="https://user-images.githubusercontent.com/62210/152285499-f5958c97-f7be-482e-a376-41a2ed827c67.png">

### After with Retype v2.0

<img width="280" alt="Screen Shot 2022-02-02 at 10 06 45 PM" src="https://user-images.githubusercontent.com/62210/152285514-f1ea7348-42ce-4fec-9a10-1bb33020773a.png">

**Comment by geoffreymcgill** [comment]
> 3. The performance of page loads are suffering quite a bit. Try to click on some items in navigation pane and note how relatively slow every page load is. We checked that and it seems that the delay is CPU bound somewhere deep inside the javascript

We are working on this performance issue right now and it will be fixed in the upcoming `Retype v2.0` release. We have made some significant performance enhancements across the UI, so things should feel much better with the next release.

**Comment by geoffreymcgill** [comment]
We also noticed a workaround that you had implemented with the `label` and using `＜＞` chars instead of `<>`. This has also been fixed in the next release. 

### Retype v1.11

```
---
label: ArrayEqualityComparer＜T＞
---
```

### Retype v2.0

```
---
label: ArrayEqualityComparer<T>
---
```
